pkg/server/api/entry/v1/service_test.go

```diff
@@ -799,6 +799,448 @@ func setupServiceTest(t *testing.T, ds datastore.DataStore) *serviceTest {
+				{
+					Ttl: 1000,
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						Ttl: 1000,
+					},
+				},
+			},
+		},
+		{
+			name:           "Success Update FederatesWith",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				FederatesWith: true,
+			},
+			outputMask: &types.EntryMask{
+				FederatesWith: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					FederatesWith: []string{},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						FederatesWith: []string{},
+					},
+				},
+			},
+		},
+		{
+			name:           "Success Update Admin",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				Admin: true,
+			},
+			outputMask: &types.EntryMask{
+				Admin: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					Admin: false,
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						Admin: false,
+					},
+				},
+			},
+		},
+		{
+			name:           "Success Update Downstream",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				Downstream: true,
+			},
+			outputMask: &types.EntryMask{
+				Downstream: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					Downstream: false,
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						Downstream: false,
+					},
+				},
+			},
+		},
+		{
+			name:           "Success Update ExpiresAt",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				ExpiresAt: true,
+			},
+			outputMask: &types.EntryMask{
+				ExpiresAt: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					ExpiresAt: 999,
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						ExpiresAt: 999,
+					},
+				},
+			},
+		},
+		{
+			name:           "Success Update DnsNames",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				DnsNames: true,
+			},
+			outputMask: &types.EntryMask{
+				DnsNames: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					DnsNames: []string{"dnsUpdated"},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						DnsNames: []string{"dnsUpdated"},
+					},
+				},
+			},
+		},
+		{
+			name:           "Success Don't Update TTL",
```

Comment:
maybe insstead of success dont update a single one you can create a test case with all false and all values updated and try it?

Reply:
Done